benchmark/err_checks.py

```diff
@@ -0,0 +1,167 @@
+"""
+Created on Tue Jun 09 12:03:52 2020
+
+@author: Henry Liu
+"""
+import tensorflow as tf
+import numpy as np
+
+
+def rel_diff(a, b):
+    """Calculate the Frobenius norm of the relative difference of the operator matrix values
+
+    Args:
+        a (Tensor([num, N, N], c64)): Array of num D(alpha) operators
+        b (Tensor([num, N, N], c64)): Array of num D(alpha) operators
+
+    Returns:
+        (float, float): (mean, max) of the relative difference
+    """
+    a = tf.cast(a, dtype=tf.complex128)
+    b = tf.cast(b, dtype=tf.complex128)
+
+    # Numerically, the values close to 0 are not going to be 0, so we need to filter
+    # out small differences that may be large compared to an underlying reference of
+    # "close to 0", i.e. 1e-8 / 1e-14 should be ignored. We can do this with
+    # np.isclose:
+    close = np.isclose(a, b, atol=1e-5)
+    a = np.where(~close, a, 0)
+    b = np.where(~close, b, 0)
+
+    # Compute relative diff
+    diffs = tf.norm(
+        tf.math.divide_no_nan(2 * tf.math.abs(a - b), tf.math.abs(a) + tf.math.abs(b)),
+        ord="fro",
+        axis=[-2, -1],
+    )
+    return tf.math.reduce_mean(diffs), tf.math.reduce_max(diffs)
+
+
+def ev_err(test):
+    """Calculate error in expectation value of <0|alpha> = exp(-1/2 * |alpha|^2)
```

Comment:
I don't think is the expectation value - it is just the coefficient of the basis state |0> in the expansion of |alpha> in Fock basis.